presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -577,15 +581,21 @@ public boolean transitionToFinishing()
         }
 
         if (autoCommit) {
-            transactionManager.asyncCommit(session.getTransactionId().get())
-                    .whenComplete((value, throwable) -> {
-                        if (throwable == null) {
-                            transitionToFinished();
-                        }
-                        else {
-                            transitionToFailed(unwrapCompletionException(throwable));
-                        }
-                    });
+            ListenableFuture<?> commitFuture = transactionManager.asyncCommit(session.getTransactionId().get());
+            Futures.addCallback(commitFuture, new FutureCallback<Object>()
+            {
+                @Override
+                public void onSuccess(@Nullable Object result)
+                {
+                    transitionToFinished();
+                }
+
+                @Override
+                public void onFailure(Throwable throwable)
+                {
+                    transitionToFailed(unwrapCompletionException(throwable));
+                }
+            });
```

Comment:
This example shows how the `ListenableFuture` API gives you

* more verbose code, as the API is not lambda-friendly
* less fluent code (as it's based on static methods)

More verbose code is harder to read and thus more prone to errors. The conversion from one futures API to another is advertised as reducing risk of errors. Could you please help me understand how LFs are less bug-prone  when compared with CFs ?